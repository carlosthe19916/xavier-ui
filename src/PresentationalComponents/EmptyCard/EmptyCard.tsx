import React from 'react';
import { Card, CardBody, Bullseye, EmptyState, EmptyStateVariant, EmptyStateIcon, Title } from '@patternfly/react-core';
import { InfoIcon } from '@patternfly/react-icons';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Card/card';
import titleStyles from '@patternfly/react-styles/css/components/Title/title';

interface Props {
    title: string | React.ReactElement;
    minHeight?: number;
}

export const EmptyCard: React.FC<Props> = ({ title, minHeight }) => {
    return (
        <Card>
            <div className={css(styles.cardHeader, titleStyles.title, titleStyles.modifiers.xl)}>{title}</div>
            <CardBody style={minHeight ? { height: minHeight } : undefined}>
                <Bullseye>
                    <EmptyState variant={EmptyStateVariant.full}>
                        <EmptyStateIcon icon={InfoIcon} />
                        <Title headingLevel="h6" size="lg">
                            Not enough data to show this card.
                        </Title>
                    </EmptyState>
                </Bullseye>
            </CardBody>
        </Card>
    );
};
