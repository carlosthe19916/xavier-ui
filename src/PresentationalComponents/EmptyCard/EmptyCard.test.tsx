import React from 'react';
import { shallow } from 'enzyme';
import { EmptyCard } from './EmptyCard';

describe('EmptyCard', () => {
    it('expect to render', () => {
        const wrapper = shallow(<EmptyCard title="My Title" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('expect to render with title as Element', () => {
        const wrapper = shallow(<EmptyCard title={<h1>My Title</h1>} />);
        expect(wrapper).toMatchSnapshot();
    });
});
