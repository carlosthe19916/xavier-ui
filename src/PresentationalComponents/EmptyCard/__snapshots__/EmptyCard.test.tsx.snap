// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCard expect to render 1`] = `
<Card>
  <div
    className="pf-c-card__header pf-c-title pf-m-xl"
  >
    My Title
  </div>
  <CardBody>
    <Bullseye>
      <EmptyState
        variant="full"
      >
        <EmptyStateIcon
          icon={[Function]}
        />
        <Title
          headingLevel="h6"
          size="lg"
        >
          Not enough data to show this card.
        </Title>
      </EmptyState>
    </Bullseye>
  </CardBody>
</Card>
`;

exports[`EmptyCard expect to render with title as Element 1`] = `
<Card>
  <div
    className="pf-c-card__header pf-c-title pf-m-xl"
  >
    <h1>
      My Title
    </h1>
  </div>
  <CardBody>
    <Bullseye>
      <EmptyState
        variant="full"
      >
        <EmptyStateIcon
          icon={[Function]}
        />
        <Title
          headingLevel="h6"
          size="lg"
        >
          Not enough data to show this card.
        </Title>
      </EmptyState>
    </Bullseye>
  </CardBody>
</Card>
`;
